pkg/network/encoding/marshal/usm_http_test.go

```diff
@@ -307,6 +310,201 @@ func TestLocalhostScenario(t *testing.T) {
 	assert.Equal(uint32(1), aggregations.EndpointAggregations[0].StatsByStatusCode[int32(103)].Count)
 }
 
+// TestKubernetesLocalNATScenario tests how USMConnectionIndex handles Kubernetes-style local NAT connections
+// with both pre-NAT and post-NAT connections as seen in Kubernetes environments
+func TestKubernetesLocalNATScenario(t *testing.T) {
+	if runtime.GOOS == "windows" || os.Getenv("CI") == "true" {
+		t.Skip("Skipping test on Windows or CI")
+	}
+	assert := assert.New(t)
+
+	// Create IP addresses and ports
+	clientIP := util.AddressFromString("172.29.161.37")
+	k8sServiceIP := util.AddressFromString("10.100.103.122") // K8s Service IP (ClusterIP)
+	serverIP := util.AddressFromString("172.29.191.94")
+	clientPort := uint16(53792)
+	servicePort := uint16(7778)
+	serverPort := uint16(7777)
+
+	// Create both connections exactly as seen from SUSM-94 reproduction environment
```

Comment:
Not sure the comment with SUSM-94 is relevant  let's explain the purpose without referencing the ticket number